ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/policy-maps-qos.j2

```diff
@@ -17,3 +17,17 @@ policy-map type quality-of-service {{ policy_map }}
 {%         endfor %}
 {%     endif %}
 {% endfor %}
+{# PBR policy-map #}
+{% for policy_map in policy_maps.pbr | arista.avd.natural_sort %}
+!
+policy-map type pbr {{ policy_map }}
+{%     if policy_maps.pbr[policy_map].classes is arista.avd.defined %}
+{%         for class in policy_maps.pbr[policy_map].classes %}
+   class {{ class }}
+{%             if policy_maps.pbr[policy_map].classes[class].set.nexthop.ip_address is arista.avd.defined %}
+      set nexthop {{ 'recursive' if policy_maps.pbr[policy_map].classes[class].set.nexthop.recursive is arista.avd.defined(true) }} {{ policy_maps.pbr[policy_map].classes[class].set.nexthop.ip_address }}
```

Comment:
Please use a cli_var to generate this. See in other templates for styling.
It will make it easier to read. Also this one produces a double space if recursive is false.

Reply:
Sorry, we cant find the "cli_var"! Can you explain it? The pbr option "recursive" have no value, only "true" or "false" and the cli_set from above will not work.

Reply:
No problem, idea is to build the CLI string using all the conditional expressions and then render it.

A good example is available [here](https://github.com/titom73/ansible-avd/blob/53c032a0c1aab24080b090063d23880e737a4069/ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/mac-security.j2#L19)

```j2
{%             if mac_security.profiles[profile].connection_keys[connection_key].encrypted_key is arista.avd.defined %}
{%                 set key_cli = "key " ~ connection_key ~ " 7 " ~ mac_security.profiles[profile].connection_keys[connection_key].encrypted_key %}
{%                 if mac_security.profiles[profile].connection_keys[connection_key].fallback is arista.avd.defined %}
{%                     set key_cli = key_cli ~ " fallback" %}
{%                 endif %}
      {{ key_cli }}
{%             endif %}
```